pkg/porter/credentials.go

```diff
@@ -0,0 +1,90 @@
+package porter
+
+import (
+	"fmt"
+
+	"github.com/deislabs/porter/pkg/credentials"
+	"github.com/deislabs/porter/pkg/printer"
+	"github.com/pkg/errors"
+
+	yaml "gopkg.in/yaml.v2"
+)
+
+func (p *Porter) PrintCredentials(opts printer.PrintOptions) error {
+	return nil
+}
+
+type CredentialOptions struct {
+	sharedOptions
+	DryRun bool
+	Silent bool
+}
+
+// Validate prepares for an action and validates the options.
+// For example, relative paths are converted to full paths and then checked that
+// they exist and are accessible.
+func (g *CredentialOptions) Validate(args []string) error {
+	err := g.validateCredName(args)
+	if err != nil {
+		return err
+	}
+
+	err = g.validateBundlePath()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (g *CredentialOptions) validateCredName(args []string) error {
+	if len(args) == 1 {
+		g.Name = args[0]
+	} else if len(args) > 1 {
+		return errors.Errorf("only one positional argument may be specified, the credential name, but multiple were received: %s", args)
+	}
+	return nil
+}
+
+func (p *Porter) GenerateCredentials(opts CredentialOptions) error {
+
+	//TODO make this work for either porter.yaml OR a bundle
+	bundle, err := p.CNAB.LoadBundle(opts.File, opts.Insecure)
+	if err != nil {
+		return err
+	}
+	name := opts.Name
+	if name == "" {
+		name = bundle.Name
+	}
+	genOpts := credentials.GenerateOptions{
+		Name:        name,
+		Credentials: bundle.Credentials,
+		Silent:      opts.Silent,
+	}
+	fmt.Fprintf(p.Out, "Generating new credential %s from bundle %s\n", genOpts.Name, bundle.Name)
+	fmt.Fprintf(p.Out, "==> %d credentials required for bundle %s\n", len(genOpts.Credentials), bundle.Name)
+
+	cs, err := credentials.GenerateCredentials(genOpts)
+	if err != nil {
+		//do something with the error
```

Comment:
I vote to take it out for a steak dinner

Reply:
I think we should come up with a good way to resolve that stuff and then wire it all up. Part of me feels like we should still work with a "Bundle" object just so everything is consistent and we can easily use duffle, have the translation from porter.yaml -> Bundle (the struct) and bundle.json ->Bundle (the struct) happen in a single place and these can all jsut say context or whatever give me my bundle please and thank you. I figured we could get all this in, then go through and make the :magic: happen

Reply:
lol i wanted this to be a reply to the comment re #320

Reply:
I think we are saying the same thing? I was just referring the to the flag names and what the user sees when executing commands. 

That `--file` would be for porter.yaml and be defaulted automatically, and --cnab-file would be for bundle.json, etc ... there's more in the issue about how it should work. Then the code would go like you said from a porter.yaml to making sure the bundle.json is up to date, rebuild if necessary, then deal with the bundle from there on a single code path.

If we aren't saying the same thing, let's chat in person since this is going back and forth a bit. 😀

Reply:
> I think we are saying the same thing? I was just referring the to the flag names and what the user sees when executing commands.
 
i think we are